File: csharp/fibonacci_sequence/fibonacci_sequence_1.cs
#include <stdio.h>

// This program calculates the Fibonacci sequence up to a certain number of terms.
// The Fibonacci sequence is a series of numbers where each number is the sum of the two preceding ones.
// The sequence starts with 0 and 1, and each subsequent number is the sum of the previous two.
// This program is designed to help programmers understand how to implement the Fibonacci sequence in C.

int main() {
    int n, i;
    unsigned long long t1 = 0, t2 = 1, nextTerm;

    // Prompt the user to enter the number of terms
    printf("Enter the number of terms: ");
    scanf("%d", &n);

    // Check if the number of terms is valid
    if (n <= 0) {
        printf("Please enter a positive integer.\n");
        return 1;
    }

    // Print the first two terms of the Fibonacci sequence
    printf("Fibonacci Sequence: %llu, %llu", t1, t2);

    // Calculate the Fibonacci sequence up to the nth term
    for (i = 3; i <= n; ++i) {
        nextTerm = t1 + t2; // Calculate the next term
        printf(", %llu", nextTerm); // Print the next term
        t1 = t2; // Update t1 to the previous term
        t2 = nextTerm; // Update t2 to the current term
    }

    printf("\n");

    // Return 0 to indicate successful execution
    return 0;
}

File: vb/fibonacci_sequence/fibonacci_sequence_1.vb
' This program calculates the Fibonacci sequence up to a specified number of terms.
' The Fibonacci sequence is a series of numbers where each number is the sum of the two preceding ones.
' The sequence starts with 0 and 1, and each subsequent number is the sum of the previous two.
' This program is designed to help programmers understand how to implement the Fibonacci sequence in Visual Basic.

Module FibonacciSequence

    Sub Main()
        ' Declare variables to hold the number of terms and the first two terms of the Fibonacci sequence.
        Dim numTerms As Integer
        Dim firstTerm As Integer = 0
        Dim secondTerm As Integer = 1

        ' Prompt the user to enter the number of terms for the Fibonacci sequence.
        Console.Write("Enter the number of terms: ")
        numTerms = Convert.ToInt32(Console.ReadLine())

        ' Display the first two terms of the Fibonacci sequence.
        Console.WriteLine("Fibonacci Sequence:")
        Console.Write(firstTerm & " " & secondTerm & " ")

        ' Calculate and display the remaining terms of the Fibonacci sequence.
        For i As Integer = 3 To numTerms
            ' Calculate the next term by adding the previous two terms.
            Dim nextTerm As Integer = firstTerm + secondTerm

            ' Display the next term.
            Console.Write(nextTerm & " ")

            ' Update the first and second terms for the next iteration.
            firstTerm = secondTerm
            secondTerm = nextTerm
        Next

        ' Wait for the user to press a key before closing the console window.
        Console.WriteLine()
        Console.WriteLine("Press any key to exit.")
        Console.ReadKey()
    End Sub

End Module

